src/db.c

```diff
@@ -41,7 +41,7 @@
  * C-level DB API
  *----------------------------------------------------------------------------*/
 
-int expireIfNeeded(redisDb *db, robj *key, int force_delete_expired);
+int expireIfNeeded(redisDb *db, robj *key, int force_delete_expired, int avoid_delete_expired);
```

Comment:
maybe it's time to introduce flags?
```
#define EXPIRE_FORCE_DELETE_ON_REPLICAS (1<<0)
#define EXPIRE_AVOID_DELETE (1<<1)
expireIfNeeded(redisDb *db, robj *key, int flags);
```

Reply:
+1 for flags. It will also make the diff smaller, i.e. no change for all calls like `expireIfNeeded(db,keyobj,0)`.

Reply:
+1 for this. Change lgtm otherwise.

Reply:
i thought of that, but i didn't want to run the risk of replacing one integer with another, and some outdated code (maybe in a backport or a fork) succeeding compilation without an update.

Reply:
That's no risk IMHO. Before this change `force_delete_expired` is always 0 or 1. (`int force_delete_expired = flags & LOOKUP_WRITE && !is_ro_replica;`) If we define EXPIRE_FORCE_DELETE_ON_REPLICAS as 1 and the new flag as 2, the function works also when mixing the new function with old code.

Reply:
@zuiderkwast good plan, i'll do it.